graphene_plugin.py

```diff
@@ -764,19 +756,36 @@ def process_gql_schema(ctx: AttributeContext) -> Type:
                         )
                         continue
 
-            # Check the every resolver function has a corresponding `Field()` defintion
-            missing_field_names = set(object_info.resolvers.keys()) - set(object_info.fields.keys())
-            for name in missing_field_names:
-                if isinstance(object_info, InterfaceInfo) and name == 'type':
-                    # This is not a field resolver. It is the special `Interface` resolver that determines which
-                    # `ObjectType` to use at runtime.
+            # If no resolver function is defined, type-check the behavior of the graphene default resolver
+            if isinstance(object_info, ObjectTypeInfo):
+                # The default resolver doesn't apply to `Interface`s because the `ObjectType`s that implement them could
+                # have resolvers for their fields.
+                # TODO: Detect if any of an `Interface`'s `ObjectType`s do _not_ define their own resolver for this
+                # field. In that case, we _do_ want to type-check the default resolver.
+                fields_without_resolver_names = set(object_info.fields.keys()) - set(object_info.resolvers.keys())
+                for name in fields_without_resolver_names:
+                    gql_field = object_info.fields[name]
+                    # Note: `analyze_member_access` will call `ctx.api.fail()` if the provided type doesn't have
+                    # a member with the given name at all. So our code only needs to do the subtype check.
+                    default_resolver_return_type = analyze_member_access(
+                        gql_field.name,
+                        object_info.runtime_type,
+                        gql_field.context,
+                        False,  # is_lvalue
+                        False,  # is_super
+                        False,  # is_operator
+                        ctx.api.msg,
+                        original_type=object_info.runtime_type,
+                        chk=cast(TypeChecker, ctx.api),
+                    )
+                    if not is_subtype(default_resolver_return_type, gql_field.type):
+                        ctx.api.fail(
+                            f'Field expects type {gql_field.type} but {object_info.runtime_type}.{gql_field.name} has '
+                            f'type {default_resolver_return_type}',
+                            gql_field.context,
+                        )
```

Comment:
Then, in a separate pass, we figure out which fields don't have explicit resolvers defined (in this case we're just looking at the `ObjectType`'s own fields, not `all_fields`) and then do the default resolver type-checking on them.